std/typecons.d

```diff
@@ -565,7 +565,7 @@ if (distinctFieldNames!(Specs))
     enum areBuildCompatibleTuples(Tup1, Tup2) = isTuple!Tup2 && is(typeof(
     {
         static assert(Tup1.Types.length == Tup2.Types.length);
-        foreach (i, _; Tup1.Types)
+        static foreach (i, _; Tup1.Types)
```

Comment:
Since all that's happening in this is a static assert, is there a reason to prefer `static foreach` over `foreach`?

Reply:
Style/clarity (see e.g. https://github.com/dlang/phobos/pull/5729)